app/models/external_report.rb

```diff
@@ -43,4 +43,13 @@ def url_for(offering_id, user, protocol, host)
     "#{url}?offering=#{offering_api_url(offering_id, protocol, host)}&token=#{token}&username=#{username}"
   end
 
+  def url_for_class(class_id, user, protocol, host)
+    grant = client.updated_grant_for(user, ReportTokenValidFor)
+    token = grant.access_token
+    username = user.login
+    routes = Rails.application.routes.url_helpers
+    class_url = routes.api_v1_class_url(class_id, {protocol:protocol, host:host})
+    "#{url}?class=#{class_url}&token=#{token}&username=#{username}"
+  end
+
```

Comment:
This is going to break any existing ClassReport external reports. If a teacher clicks on the class report button, the external report won't know how to handle the `class` parameter so it will probably display nothing or an error.  As far as I know there is only class report: https://github.com/concord-consortium/tracked-question-report 
So can you update that report to handle this class param if it is present?